cli_tools/gce_vm_image_import/main.go

```diff
@@ -18,119 +18,115 @@ package main
-	project              = flag.String("project", "", "project to run in, overrides what is set in workflow")
-	scratchBucketGcsPath = flag.String("scratch_bucket_gcs_path", "", "GCS scratch bucket to use, overrides what is set in workflow")
-	oauth                = flag.String("oauth", "", "path to oauth json file, overrides what is set in workflow")
-	ce                   = flag.String("compute_endpoint_override", "", "API endpoint to override default")
-	gcsLogsDisabled      = flag.Bool("disable_gcs_logging", false, "do not stream logs to GCS")
-	cloudLogsDisabled    = flag.Bool("disable_cloud_logging", false, "do not stream logs to Cloud Logging")
-	stdoutLogsDisabled   = flag.Bool("disable_stdout_logging", false, "do not display individual workflow logs on stdout")
-	kmsKey               = flag.String("kms_key", "", "ID of the key or fully qualified identifier for the key. This flag must be specified if any of the other arguments below are specified.")
-	kmsKeyring           = flag.String("kms_keyring", "", "The KMS keyring of the key.")
-	kmsLocation          = flag.String("kms_location", "", "The Cloud location for the key.")
-	kmsProject           = flag.String("kms_project", "", "The Cloud project for the key")
-	noExternalIP         = flag.Bool("no_external_ip", false, "VPC doesn't allow external IPs")
-	labels               = flag.String("labels", "", "List of label KEY=VALUE pairs to add. Keys must start with a lowercase character and contain only hyphens (-), underscores (_), lowercase characters, and numbers. Values must contain only hyphens (-), underscores (_), lowercase characters, and numbers.")
-	storageLocation      = flag.String("storage_location", "", "Location for the imported image which can be any GCS location. If the location parameter is not included, images are created in the multi-region associated with the source disk, image, snapshot or GCS bucket.")
-	uefiCompatible       = flag.Bool("uefi_compatible", false, "Enables UEFI booting, which is an alternative system boot method. Most public images use the GRUB bootloader as their primary boot method.")
-	sysprepWindows       = flag.Bool("sysprep_windows", false, "Whether to generalize image using Windows Sysprep. Only applicable to Windows.")
-)
+func main() {
+	log.SetPrefix("[import-image] ")
+	ctx := context.Background()
 
-func importEntry() (service.Loggable, error) {
+	importArgs, err := parseAllArgs(ctx)
+	if err != nil {
+		terminate(err)
+	}
 
-	ctx := context.Background()
-	metadataGCE := &compute.MetadataGCE{}
+	importerClosure := func() (service.Loggable, error) {
+		wf, e := importer.NewImporter(importArgs.Env, importArgs.Img, importArgs.Translation).Run(ctx)
+		return service.NewLoggableFromWorkflow(wf), e
+	}
+
+	project := importArgs.Env.Project
+	if err := service.RunWithServerLogging(
+		service.ImageImportAction, initLogging(importArgs), &project, importerClosure); err != nil {
```

Comment:
this returned value of initLogging needs to be used by all calling to RunWithServerLogging. Try to either init it early and use it everywhere, or call RunWithServerLogging in only one place

Reply:
There are two paths to RunWithServerLogging -- one via terminate, which occurs when there's a failure during flag processing, and one via the normal path that includes import.